lib/numorstring/protocol.go

```diff
@@ -48,6 +48,23 @@ func ProtocolFromInt(p uint8) Protocol {
 	)
 }
 
+func ProtocolV3FromProtocolV1(p Protocol) Protocol {
+	if p.Type == NumOrStringNum {
+		return p
+	}
+
+	for _, n := range allProtocolNames {
+		if strings.ToLower(n) == strings.ToLower(p.StrVal) {
+			return Protocol(
+				Uint8OrString{Type: NumOrStringString, StrVal: n},
+			)
+		}
+	}
+
+	// Unknown protocol. TODO: return error?
```

Comment:
I think we're fine, no need to return the error, should've been caught by the v1 validator